api/src/routes/user.ts

```diff
@@ -90,6 +106,119 @@ export const userRoutes: FastifyPluginCallbackTypebox = (
       }
     }
   );
+  fastify.get(
+    '/user/get-session-user',
+    {
+      schema: schemas.getSessionUser
+    },
+    async (req, res) => {
+      try {
+        // TODO: promise all to fetch both
+        const userToken = await fastify.prisma.userToken.findFirst({
+          where: { userId: req.session.user.id }
+        });
+
+        const encodedToken = userToken
+          ? encodeUserToken(userToken.id)
+          : undefined;
+
+        console.log('encodedToken', encodedToken);
+
+        const user = await fastify.prisma.user.findUnique({
+          where: { id: req.session.user.id },
+          select: {
+            about: true,
+            acceptedPrivacyTerms: true,
+            completedChallenges: true,
+            currentChallengeId: true,
+            donationEmails: true,
+            email: true,
+            emailVerified: true,
+            githubProfile: true,
+            id: true,
+            is2018DataVisCert: true,
+            isApisMicroservicesCert: true,
+            isBackEndCert: true,
+            isCheater: true,
+            isCollegeAlgebraPyCertV8: true,
+            isDataAnalysisPyCertV7: true,
+            isDataVisCert: true,
+            isDonating: true,
+            isFrontEndCert: true,
+            isFrontEndLibsCert: true,
+            isFullStackCert: true,
+            isHonest: true,
+            isInfosecCertV7: true,
+            isInfosecQaCert: true,
+            isJsAlgoDataStructCert: true,
+            isMachineLearningPyCertV7: true,
+            isQaCertV7: true,
+            isRelationalDatabaseCertV8: true,
+            isRespWebDesignCert: true,
+            isSciCompPyCertV7: true,
+            keyboardShortcuts: true,
+            linkedin: true,
+            location: true,
+            name: true,
+            partiallyCompletedChallenges: true,
+            picture: true,
+            portfolio: true,
+            profileUI: true,
+            progressTimestamps: true,
+            savedChallenges: true,
+            sendQuincyEmail: true,
+            sound: true,
+            theme: true,
+            twitter: true,
+            username: true,
+            usernameDisplay: true,
+            website: true,
+            yearsTopContributor: true
+          }
+        });
+
+        if (!user?.username) {
+          void res.code(500);
+          return { user: {}, result: '' };
+        }
+
+        const {
+          usernameDisplay,
+          completedChallenges,
+          progressTimestamps,
+          twitter,
+          ...publicUser
+        } = user;
+
+        return {
+          user: {
+            [user.username]: {
+              ...removeNulls(publicUser),
+              completedChallenges: completedChallenges.map(removeNulls),
+              completedChallengeCount: completedChallenges.length,
```

Comment:
I removed this, because the client does not even use it.

Reply:
Yes 🙈 I misspelt it as `conpletedChallengeCount` and was searching that - it is not used anywhere.